app/gui/src/project-view/util/autoBlur.ts

```diff
@@ -138,3 +138,33 @@ function handleClick(
   }
   return wrappedInteraction
 }
+
+const blurHandlers = new Set<() => void>()
+
+/**
+ * Register a callback to be called when the window is blurred.
+ * The callback will be called when the window is blurred, and will be removed when the component is unmounted.
+ */
+export function onBlur(callback: () => void) {
+  blurHandlers.add(callback)
+  onUnmounted(() => {
+    blurHandlers.delete(callback)
+  })
+}
+
```

Comment:
It's the same as `useEvent(window, 'blur')`, isn't it? Because `useEvent` also removes itself (on scope disposal, but it's effectively same as unmount).

Reply:
Yes, but here we have a single event handler per application instead of multiple event handlers per node. We use the same pattern for autoBlur, for performance reasons.

Reply:
Didn't know that handling a set of callbacks "by hand" is faster than multiple listeners (you may consider adding a note to someone as ill-informed as me)